### ISSUE TITLE
[Bug]: Issue with aws_dms_replication_task resource replication_task_settings Logging.CloudWatchLogStream is read-only and cannot be set

### ISSUE THREAD

**Issue by Volodymyr-Kuchinskyi** [hashicorp/terraform-provider-aws#36997]
### Terraform Core Version

v1.4.0

### AWS Provider Version

5.46.0

### Affected Resource(s)

* aws_dms_replication_task

### Expected Behavior

Plan running without any errors

### Actual Behavior

The parameter Logging.CloudWatchLogStream is read-only and cannot be set.

### Relevant Error/Panic Output Snippet

```shell
Errors:

in resource "aws_dms_replication_task" "dms_task":
replication_task_settings = local.task_settings

The parameter Logging.CloudWatchLogStream is read-only and cannot be set.

Error: Invalid value
in resource "aws_dms_replication_task" "dms_task":
replication_task_settings = local.task_settings

The parameter Logging.CloudWatchLogGroup is read-only and cannot be set.

Error: Invalid value
```

### Terraform Configuration Files

```
resource "aws_dms_replication_task" "dms_task" {
  replication_task_id       = local.task_id
  replication_instance_arn  = var.instance_arn
  migration_type            = "full-load-and-cdc"
  replication_task_settings = local.task_settings
  source_endpoint_arn       = var.source_endpoint_arn // postgres endpoint
  target_endpoint_arn       = var.target_endpoint_arn // postgres endpoint
  table_mappings            = local.table_mappings
  lifecycle {
    ignore_changes = [replication_task_settings]
  }
  tags = merge(
    var.account.account_tags,
    {
      Name = local.task_id
    }
  )
}
```

### Steps to Reproduce

1) Create a DMS task with version 5.45.0
2) Try to run terraform plan with enabled and disabled logging for dms task

### Important Factoids

The dms task was created ~2 years ago with a different version of terraform provider
`terraform plan` runs without any errors for `5.45.0` so there is something introduced in `5.46.0`

### References

#36936

### Would you like to implement a fix?

None

**Comment by Rohan8782git** [comment]
The parameter Logging.CloudWatchLogGroup is read-only and cannot be set.

Didn't work with aws provider 5.46.0 is there any update on this?

**Comment by ewok** [comment]
CloudWatchLogGroup and CloudWatchLogStream seem to be dynamic and changed by aws itself.
Try to remove these two from your replication_task_settings.
This worked for me.

**Comment by zachary-povey** [comment]
From the release notes: 
![image](https://github.com/hashicorp/terraform-provider-aws/assets/64191599/c340ffd6-f808-4629-a1fe-7b3f96f15978)

We've just hit this too as our settings had these explicitly set to null (presumably we had this due to a copy and paste from the docs [here](https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Tasks.CustomizingTasks.TaskSettings.Logging.html)) so yeah think it's just a case of removing these settings entirely.

**Comment by eugenetaranov** [comment]
seems like it was already fixed in v.3.36 https://github.com/hashicorp/terraform-provider-aws/pull/13476 but then can be replicated with 5.10+

**Comment by YakDriver** [comment]
We apologize for the inconvience caused by this!

TL;DR This change was required due to an AWS change. The fix is to remove `Logging.CloudWatchLogGroup` and `Logging.CloudWatchLogStream` from which are read only anyway.

### Correct Example

`EnableLogging` is set. `EnableLogContext` could also be set here since `EnableLogging` is true.

```terraform
resource "aws_dms_replication_task" "test" {
  # ... other config ...

  replication_task_settings = jsonencode(
    {
      Logging = {
        EnableLogging = true,
        LogComponents = [{
          Id       = "DATA_STRUCTURE",
          Severity = "LOGGER_SEVERITY_WARNING"
        }]
      }
    }
  )
}
```

### Incorrect Example 1

`EnableLogging` is set but regardless `CloudWatchLogGroup` and `CloudWatchLogStream` are read only and cannot be set.

```terraform
resource "aws_dms_replication_task" "test" {
  # ... other config ...

  replication_task_settings = jsonencode(
    {
      Logging = {
        EnableLogging = true,
        CloudWatchLogGroup = "abcdef"
        CloudWatchLogStream = "abcdef"
      }
    }
  )
}
```

### Incorrect Example 2

`EnableLogging` is set but regardless `CloudWatchLogGroup` and `CloudWatchLogStream` are read only and shouldn't be set.

```terraform
resource "aws_dms_replication_task" "test" {
  # ... other config ...

  replication_task_settings = jsonencode(
    {
      Logging = {
        EnableLogging = true,
        CloudWatchLogGroup = null
        CloudWatchLogStream = null
      }
    }
  )
}
```

### Background

Unfortunately, our hand was forced by a change made by AWS. Previously, `Logging.CloudWatchLogGroup` and `Logging.CloudWatchLogStream` could be sent to AWS even though they did nothing, and AWS would not complain. However, AWS made a change so that including them would result in constant diffs (see #36598). In reality, AWS fixed the behavior because these settings are read only and cannot be changed. [You can enable](https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Tasks.CustomizingTasks.TaskSettings.Logging.html) logging with `Logging.EnableLogging` but then AWS sets `Logging.CloudWatchLogGroup` and `Logging.CloudWatchLogStream` so they should not be included in the configuration.
